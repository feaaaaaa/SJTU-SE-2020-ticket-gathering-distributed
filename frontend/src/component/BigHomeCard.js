import { Card } from 'antd';
import React from "react";
import "../css/BigHomeCard.css"
import {Link} from "react-router-dom";

const { Meta } = Card;

export class BigHomeCard extends React.Component{
    componentDidMount() {
        console.log(this.props.info.actitems);
    }

    render(){
        return(
            <Link to="/sortPage" onClick={()=>localStorage.setItem("cardInfo",JSON.stringify(this.props.info))}>
            <Card  bordered={false} hoverable={true} style={{height:420}}>
                <img alt="example" src={this.props.info.activityIcon} style={{width:180,height:260,float:"none"}}/>
                <div>
                    <p className="Box">{this.props.info.title}</p>
                    <p id="Venue">{this.props.info.venue}</p>
                    <p id="Venue">{this.props.info.timescale}</p>
                    <p id="Pricestr">{'Â¥'+this.props.info.actitems[0].price[0].class[0].price+'-'
                    +this.props.info.actitems[0].price[0].class[this.props.info.actitems[0].price[0].class.length-1].price}</p>
                </div>
            </Card>
            </Link>
        )
    }
}