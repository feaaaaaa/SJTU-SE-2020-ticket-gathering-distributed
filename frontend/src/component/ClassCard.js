import { Card,Collapse,Menu, Dropdown, Button } from 'antd';
import React from "react";

export class ClassCard extends React.Component{
    constructor(props) {
        super(props);
    }

    render() {
        const menu1 = (
            <Menu>
                <Menu.Item key="1">展会</Menu.Item>
                <Menu.Item key="2">特色体验</Menu.Item>
                <Menu.Item key="3">其他展览休闲</Menu.Item>
            </Menu>
        );
        const menu2 = (
            <Menu>
                <Menu.Item key="1">话剧</Menu.Item>
                <Menu.Item key="2">音乐剧</Menu.Item>
                <Menu.Item key="3">其他话剧歌剧</Menu.Item>
            </Menu>
        );
        const menu3 = (
            <Menu>
                <Menu.Item key="1">摇滚</Menu.Item>
                <Menu.Item key="2">流行</Menu.Item>
                <Menu.Item key="3">音乐节</Menu.Item>
                <Menu.Item key="4">其他演唱会</Menu.Item>
            </Menu>
        );
        const menu4 = (
            <Menu>
                <Menu.Item key="1">室内乐及古乐</Menu.Item>
                <Menu.Item key="2">独奏</Menu.Item>
                <Menu.Item key="3">管弦乐</Menu.Item>
                <Menu.Item key="4">其他音乐会</Menu.Item>
            </Menu>
        );
        const menu5 = (
            <Menu>
                <Menu.Item key="1">戏曲</Menu.Item>
                <Menu.Item key="2">相声</Menu.Item>
                <Menu.Item key="3">魔术</Menu.Item>
                <Menu.Item key="4">其他曲苑杂坛</Menu.Item>
            </Menu>
        );
        const menu6 = (
            <Menu>
                <Menu.Item key="1">球类运动</Menu.Item>
                <Menu.Item key="2">田径</Menu.Item>
                <Menu.Item key="3">篮球</Menu.Item>
                <Menu.Item key="4">足球</Menu.Item>
                <Menu.Item key="5">其他体育</Menu.Item>
            </Menu>
        );
        const menu7 = (
            <Menu>
                <Menu.Item key="1">舞蹈</Menu.Item>
                <Menu.Item key="2">其他舞蹈芭蕾</Menu.Item>
            </Menu>
        );
        const menu8 = (
            <Menu>
                <Menu.Item key="1">其他儿童亲子</Menu.Item>
            </Menu>
        );
        return(
            <div>
                <Dropdown.Button type="primary" overlay={menu1}>展览休闲</Dropdown.Button>
                &nbsp;&nbsp;&nbsp;
                <Dropdown.Button type="primary" overlay={menu2}>话剧歌剧</Dropdown.Button>
                &nbsp;&nbsp;&nbsp;
                <Dropdown.Button type="primary" overlay={menu3}>演唱会</Dropdown.Button>
                &nbsp;&nbsp;&nbsp;
                <Dropdown.Button type="primary" overlay={menu4}>音乐会</Dropdown.Button>
                &nbsp;&nbsp;&nbsp;
                <Dropdown.Button type="primary" overlay={menu5}>曲苑杂坛</Dropdown.Button>
                &nbsp;&nbsp;&nbsp;
                <Dropdown.Button type="primary" overlay={menu6}>体育</Dropdown.Button>
                &nbsp;&nbsp;&nbsp;
                <Dropdown.Button type="primary" overlay={menu7}>舞蹈芭蕾</Dropdown.Button>
                &nbsp;&nbsp;&nbsp;
                <Dropdown.Button type="primary" overlay={menu8}>儿童亲子</Dropdown.Button>
            </div>
        )
    }
}
