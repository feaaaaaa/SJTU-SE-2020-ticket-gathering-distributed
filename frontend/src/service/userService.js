import {postRequest,fetchPost1} from "../utils/ajax";

export const checkUser = (value,callback) => {
    const data={username:value};
    const url = `http://localhost:8080/ExistsByUsername`;
    postRequest(url, data, callback);
};


export const login = (value,callback) => {
    const data={username:value.username,password:value.password}
    const url = 'http://localhost:8080/Login';
    fetchPost1(url, data, callback);
};

export const register = (username,password,email,personicon,phone,gender,callback) => {
    const url = 'http://localhost:8080/Register';
    const data={
        username:username,
        password:password,
        type:"User",
        gender: gender,
        email:email,
        phone:phone,
        personIcon:personicon
    };
    postRequest(url, data, callback);
};

// export const register = (self) => {
//     const url = 'http://localhost:8080/User/Register';
//     const callback = (data) => {
//         if(data.status >= 0) {
//             localStorage.setItem('userId', data.data.userId);
//             localStorage.setItem('userName', data.data.username);
//             self.$message({
//                 message: "Register successfully",//data.msg,
//                 type: "success",
//             });
//             self.$router.replace("/");
//         }
//         else{
//             //message.error(data.msg);
//             self.$message({
//                 message: "Username is repetitive! Please choose another one.",//data.msg,
//                 type: "error",
//             });
//         }
//     };
//     let user={username:self.username,password:self.password,type:"User",gender: "Male",email:self.email,phone:self.phone,personIcon:image2Base64(self.personicon)};
//     postRequest(url, user, callback);
// };

// function image2Base64(img) {
//     var canvas = document.createElement("canvas");
//     canvas.width = img.width;
//     canvas.height = img.height;
//     var ctx = canvas.getContext("2d");
//     ctx.drawImage(img, 0, 0, img.width, img.height);
//     var dataURL = canvas.toDataURL("image/png");
//     return dataURL;
// }