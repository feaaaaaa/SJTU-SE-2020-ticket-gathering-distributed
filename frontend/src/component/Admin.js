import React, { useState, useEffect, useRef } from 'react';
import { Form, Input, InputNumber, Modal, Button, Avatar, Typography,DatePicker } from 'antd';
import { SmileOutlined, UserOutlined } from '@ant-design/icons';
import "../css/admin.css"
import {addActivity} from "../service/AdminService";

const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 16 },
};
const tailLayout = {
    wrapperCol: { offset: 8, span: 16 },
};

// interface ModalFormProps {
//     visible: boolean;
//     onCancel: () => void;
// }

// reset form fields when modal is form, closed
const useResetFormOnCloseModal = ({ form, visible }) => {
    const prevVisibleRef = useRef();
    useEffect(() => {
        prevVisibleRef.current = visible;
    }, [visible]);
    const prevVisible = prevVisibleRef.current;
    useEffect(() => {
        if (!visible && prevVisible) {
            form.resetFields();
        }
    }, [visible]);
};

const ModalForm = ({ daycnt,classcnt,visible, onCancel }) => {
    let i;
    const [form] = Form.useForm();

    useResetFormOnCloseModal({
        form,
        visible,
    });

    const onOk = () => {
        form.submit();
    };

    const item = [];
    item.push(
        <Form form={form} layout="vertical" name="userForm">
            <Form.Item name="website"
                       label="website"
                       rules={[{required: true}]}
            >
                <Input/>
            </Form.Item>
        </Form>
    );

    for (i = 0; i < classcnt*daycnt; i++) {
        if(i%classcnt===0){
            item.push(
                <Form form={form} layout="vertical" name="userForm">
                    <Form.Item name={"day" + (Math.floor(i / classcnt) + 1)}
                               label={"day" + (Math.floor(i / classcnt) + 1)}
                               rules={[{required: true}]}
                    >
                        <Input/>
                        {/*<DatePicker style={{ width: '50%' }} />*/}
                    </Form.Item>
                </Form>
            )
        }
        item.push(
            <Form form={form} layout="vertical" name="userForm">
                <Form.Item name={"day"+(Math.floor(i/classcnt)+1)+"class"+(i%classcnt+1)}
                           label={"day"+(Math.floor(i/classcnt)+1)+"class"+(i%classcnt+1)}
                           rules={[{required: true}]}>
                    <InputNumber/>
                </Form.Item>
            </Form>
        );
    }


    return (
        <Modal title="Basic Drawer" visible={visible} onOk={onOk} onCancel={onCancel}>
            {/*<Form form={form} layout="vertical" name="userForm">*/}
            {/*    <Form.Item name="name" label="User Name" rules={[{ required: true }]}>*/}
            {/*        <Input />*/}
            {/*    </Form.Item>*/}
            {/*    <Form.Item name="age" label="User Age" rules={[{ required: true }]}>*/}
            {/*        <InputNumber />*/}
            {/*    </Form.Item>*/}
            {/*</Form>*/}
            {item}
        </Modal>
    );
};
const Demo = () => {
    const [visible, setVisible] = useState(false);

    const [u, setU] = useState([]);

    const [daycnt, setDaycnt] = useState(0);

    const [classcnt, setClasscnt] = useState(0);

    const showUserModal = () => {
        setVisible(true);
    };

    const hideUserModal = () => {
        setVisible(false);
    };

    const onFinish = values => {
        console.log('Finish:', values);
        console.log('users:',u);
        // let activity=[];
        // activity.push(values);
        // activity.push(u);
        var arr = [];
        arr.push(u.length);
        arr.push(daycnt);
        arr.push(classcnt);
        for (let i in values) {
            arr.push(values[i]); //属性
            //arr.push(obj[i]); //值
        }
        for(let i in u){
            for(let j in u[i]){
                arr.push(u[i][j])
            }
        }
        // arr=arr.concat(u);
        console.log("activity:"+JSON.stringify(arr));
        addActivity(JSON.stringify(arr),(res)=>{console.log(res)})
    };

    const onFinishcnt = values => {
        console.log('Success:', values);
        console.log(classcnt);
        console.log(daycnt);
        setClasscnt(values.classcnt);
        setDaycnt(values.daycnt);
    };



    return (
        <>
            <Form
                {...layout}
                name="cnt"
                initialValues={{ remember: true }}
                onFinish={onFinishcnt}
            >
                <Form.Item
                    label="day count"
                    name="daycnt"
                    rules={[{ required: true}]}
                >
                    <InputNumber min={1} max={10}/>
                </Form.Item>

                <Form.Item
                    label="class count"
                    name="classcnt"
                    rules={[{ required: true}]}
                >
                    <InputNumber min={1} max={5}/>
                </Form.Item>

                <Form.Item {...tailLayout}>
                    <Button type="primary" htmlType="submit">
                        SET CNT
                    </Button>
                </Form.Item>
            </Form>

            {/*{value1}*/}
            <Form.Provider
                onFormFinish={(name, { values, forms }) => {
                    if (name === 'userForm') {
                        const { basicForm } = forms;
                        let users = basicForm.getFieldValue('users') || [];
                        // console.log("users:"+JSON.stringify(users));
                        // console.log("value:"+JSON.stringify(values));
                        basicForm.setFieldsValue({ users: [...users, values] });
                        users=basicForm.getFieldValue('users') || [];
                        console.log("usersNew:"+JSON.stringify(users));
                        setVisible(false);
                        setU(users);
                    }
                }}
            >
                <Form {...layout} name="basicForm" onFinish={onFinish}>

                    <Form.Item name="title" label="title" rules={[{ required: true }]}>
                        <Input />
                    </Form.Item>

                    <Form.Item name="actor" label="actor" rules={[{ required: true }]}>
                        <Input />
                    </Form.Item>

                    <Form.Item name="timescale" label="timescale" rules={[{ required: true }]}>
                        <Input />
                    </Form.Item>

                    <Form.Item name="venue" label="venue" rules={[{ required: true }]}>
                        <Input />
                    </Form.Item>

                    <Form.Item
                        // rules={[{ required: true }]}
                        // name="users"
                        label="Website List"
                        shouldUpdate={(prevValues, curValues) => prevValues.users !== curValues.users}
                        // shouldUpdate={true}
                    >
                        {({ getFieldValue }) => {
                            const users = getFieldValue('users') || [];
                            console.log("user:"+JSON.stringify(users));

                            return users.length ? (
                                <ul>
                                    {users.map((user, index) => (
                                        <li key={index} className="user">
                                            {/*<Avatar icon={<UserOutlined />} />*/}
                                            {user.website}
                                            {/*- {user.day}*/}
                                        </li>
                                    ))}
                                </ul>
                            ) : (
                                <Typography.Text className="ant-form-text" type="secondary">
                                    ( <SmileOutlined /> No website yet. )
                                </Typography.Text>
                            );
                        }}
                    </Form.Item>

                    <Form.Item {...tailLayout}>
                        <Button htmlType="submit" type="primary">
                            Submit
                        </Button>
                        <Button htmlType="button" style={{ margin: '0 8px' }} onClick={showUserModal}>
                            Add Website
                        </Button>
                    </Form.Item>

                </Form>

                <ModalForm visible={visible} onCancel={hideUserModal} daycnt={daycnt} classcnt={classcnt}/>
            </Form.Provider>
        </>
    );
};
export class Admin extends React.Component{

    render() {
        return (
            <Demo/>
        );
    }
}
// export class Admin extends React.Component{
//
//     render() {
//         return (
//             <Form name="dynamic_form_item" {...formItemLayoutWithOutLabel} onFinish={onFinish}>
//                 <Form.List name="names">
//                     {(fields, { add, remove }) => {
//                         return (
//                             <div>
//                                 {fields.map((field, index) => (
//                                     <Form.Item
//                                         {...(index === 0 ? formItemLayout : formItemLayoutWithOutLabel)}
//                                         label={index === 0 ? 'Passengers' : ''}
//                                         required={false}
//                                         key={field.key}
//                                     >
//                                         <Form.Item
//                                             {...field}
//                                             validateTrigger={['onChange', 'onBlur']}
//                                             rules={[
//                                                 {
//                                                     required: true,
//                                                     whitespace: true,
//                                                     message: "Please input passenger's name or delete this field.",
//                                                 },
//                                             ]}
//                                             noStyle
//                                         >
//                                             <Input placeholder="passenger name" style={{ width: '60%' }} />
//                                         </Form.Item>
//                                         {fields.length > 1 ? (
//                                             <MinusCircleOutlined
//                                                 className="dynamic-delete-button"
//                                                 style={{ margin: '0 8px' }}
//                                                 onClick={() => {
//                                                     remove(field.name);
//                                                 }}
//                                             />
//                                         ) : null}
//                                     </Form.Item>
//                                 ))}
//                                 <Form.Item>
//                                     <Button
//                                         type="dashed"
//                                         onClick={() => {
//                                             add();
//                                         }}
//                                         style={{ width: '60%' }}
//                                     >
//                                         <PlusOutlined /> Add field
//                                     </Button>
//                                 </Form.Item>
//                             </div>
//                         );
//                     }}
//                 </Form.List>
//
//                 <Form.Item>
//                     <Button type="primary" htmlType="submit">
//                         Submit
//                     </Button>
//                 </Form.Item>
//             </Form>
//         );
//     }
//
// }