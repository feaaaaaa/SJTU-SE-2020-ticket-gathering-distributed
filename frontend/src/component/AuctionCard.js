import { Card,Collapse,Menu, Dropdown, Button,Radio, notification, Space,InputNumber,Modal } from 'antd';
import React from "react";
import "../css/Auction.css"
import {addOrder} from "../service/orderServcie";
import moment from "moment";
import {joinAuctions} from "../service/AuctionService";

export class AuctionCard extends React.Component{

    constructor(props) {
        super(props);
        this.state={
            visible:false,
            price:this.props.info.price,
        }
    }

    showModal = () => {
        this.setState({
            visible: true,
        });
    };

    handleOk = e => {
        if(this.props.info.price >= this.state.price)
            this.openNotificationLowerPrice('warning')
        else{
            const callback = data => {
                if(data === -1)
                    this.openNotificationIsOver("warning");
                else {
                    this.openNotificationPurchase("success");
                }
                this.setState({visible: false});
                this.props.getRenderCallback(1);
            }
            joinAuctions(this.props.info.auctionid,parseInt(localStorage.getItem("userId")),this.state.price,localStorage.getItem("token"),callback);
        }
    };

    handleCancel = e => {
        console.log(e);
        this.setState({
            visible: false,
        });
    };

    onChange = value => {
        this.setState({price:value});
        console.log('changed', value);
    }

    openNotificationWithoutLogin = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '未登录，请先登录',
        });
    };

    openNotificationLowerPrice = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '出价必须高于当前价格',
        });
    };

    openNotificationIsOver = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '拍卖已结束',
        });
    };

    openNotificationPurchase = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '拍卖成功',
        });
    };

    // AuctionButton = () => {
    //     if(localStorage.getItem("userId") === null)
    //         return <Button type="primary" onClick={this.openNotificationWithoutLogin("warning")}>未登录</Button>
        // else if (this.props.info.userid === parseInt(localStorage.getItem("userId"))) {
        //     return <Button type="primary" danger onClick={this.showModal}>当前最高，可继续加价</Button>;
        // }
        // return <Button type="primary" onClick={this.showModal}>竞价</Button>;
    // }


    render() {
        // const {AuctionButton} = this.AuctionButton;
        const AuctionButtion = localStorage.getItem("userId") === null ? (
        <Button type="primary" onClick={this.openNotificationWithoutLogin("warning")}>未登录</Button>
        ):(this.props.info.userid === parseInt(localStorage.getItem("userId"))?(
        <Button type="primary" danger onClick={this.showModal}>当前最高，可继续加价</Button>
            ):(<Button type="primary" onClick={this.showModal}>竞价</Button>));
        return(
            <div>
                <img className='A-image' alt="example" src={this.props.info.activityIcon} />
                <p id="A-title">{this.props.info.title}</p>
                <p id="A-info">表演者:{this.props.info.actor}</p>
                <p id="A-info">地点:{this.props.info.venue}</p>
                <p id="A-info">活动时间:{this.props.info.showtime.substr(0,19).replace("T"," ")}</p>
                <p id="A-info">截止日期:{this.props.info.ddl.substr(0,19).replace("T"," ")}</p>
                <p id="A-info">票数:{this.props.info.amount}</p>
                <p id="A-price">当前价格:{'¥'+this.props.info.price}</p>
                <div id="A-button-align">
                    {AuctionButtion}
                    {/*<AuctionButton/>*/}
                    {/*<Button>竞价</Button>*/}
                </div>
                <Modal
                    title="确认竞价"
                    visible={this.state.visible}
                    onOk={this.handleOk}
                    onCancel={this.handleCancel}
                >
                    <InputNumber
                        defaultValue={this.props.info.price}
                        formatter={value => `$ ${value}`.replace(/\B(?=(\d{3})+(?!\d))/g, ',')}
                        parser={value => value.replace(/\$\s?|(,*)/g, '')}
                        onChange={this.onChange}
                    />
                </Modal>
            </div>
        )
    }
}