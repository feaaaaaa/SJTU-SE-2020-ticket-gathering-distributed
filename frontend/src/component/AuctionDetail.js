import React from "react";
import {Button, Card, InputNumber, message, Modal, notification, Radio} from "antd";
import "../css/Detail.css"
import {getPriceById, joinAuctions} from "../service/AuctionService";


export class AuctionDetail extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            activity:[],
            visible:false,
            clickAuction:true,
            price:0,
            interval:1000,
            updateInterval: null// 定时器标识，清除用
        }
    }
    
    componentDidMount() {
        const auctionInfo=localStorage.getItem("auctionInfo");
        console.log("auctionInfo:");
        console.log(auctionInfo);
        let parsedData;

        if(auctionInfo!=null){
            parsedData=JSON.parse(auctionInfo);
            this.setState({
                activity:parsedData
            })
            this.setState({
                price:parsedData.price
            })
        }
        const Id=parsedData.auctionid;
        const token=localStorage.getItem("token");
        let errorCnt=0;
        this.state.updateInterval=setInterval(()=>{
            getPriceById(Id,token,(res)=>{
                console.log("fresh:");
                console.log(res);
                if(res!=null){
                    if(res.status===200){
                        let auctionInfo=this.state.activity;
                        auctionInfo.price=res.data;
                        this.setState({activity:auctionInfo});
                    }
                    else{
                        errorCnt++;
                        message.error(res.msg);
                        if(errorCnt>3)
                            this.setState({interval:2000});
                        if(errorCnt>10)
                            this.setState({interval:60*1000});
                    }
                }
            })
        },this.state.interval)
    }

    componentWillUnmount() {
        clearInterval(this.state.updateInterval);
    }

    onChange = value => {
        this.setState({price:value});
        console.log('changed', value);
    }

    openNotificationWithoutLogin = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '未登录，请先登录',
        });
    };

    openNotificationLowerPrice = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '出价必须高于当前价格',
        });
    };

    openNotificationWithoutEnoughMoney = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '余额不足，请充值',
        });
    };

    openNotificationIsOver = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '拍卖已结束',
        });
    };

    openNotificationPurchase = type => {
        notification[type]({
            message: 'Notification Title',
            description:
                '拍卖成功',
        });
    };

    showModal = () => {
        this.setState({
            visible: true,
        });
    };


    handleOk = () => {
        if(this.state.activity.price >= this.state.price)
            this.openNotificationLowerPrice('warning');
        else if(this.state.clickAuction){//Todo
            this.setState({clickAuction:false});
            const callback = data => {
                if (data != null) {
                    if (data.data === -1)
                        this.openNotificationIsOver("warning");
                    else if (data.data === -2)
                        this.openNotificationLowerPrice("warning");
                    else if (data.data === -3)
                        this.openNotificationWithoutEnoughMoney("warning");
                    else if(data.status===200){
                        this.openNotificationPurchase("success");
                    }else message.error(data.msg);
                    this.setState({visible: false});
                    // this.props.getRenderCallback(1);
                }
            }
            joinAuctions(this.state.activity.auctionid,parseInt(localStorage.getItem("userId")),this.state.price,localStorage.getItem("token"),callback);
            setTimeout(()=>{this.setState({clickAuction:true})}, 2000);
        }else message.error("点击太快了，休息一会吧")
    };

    handleCancel = () => {
        this.setState({
            visible: false,
        });
    };
    
    render() {
        const AuctionButtion = localStorage.getItem("userId") === null ? (
            <Button type="primary" onClick={this.openNotificationWithoutLogin("warning")}>未登录</Button>
        ):(<Button type="primary" onClick={this.showModal}>出价</Button>);
        return (
            <div>
                <img id='Dimg' alt="example" src={this.state.activity.activityIcon} />
                <p id="Dtitle">{this.state.activity.title}</p>
                <p id="Dinfo">地点:{this.state.activity.venue}</p>
                <p id="Dinfo">表演者:{this.state.activity.actor}</p>
                <p id="Dinfo">数量:{this.state.activity.amount}</p>
                <p id="Dinfo">演出时间:{this.state.activity.showtime}</p>
                <p id="Dinfo">竞拍截止时间:{this.state.activity.ddl}</p>
                <p className="price">目前价格: {this.state.activity.price}</p>
                <div style={{paddingLeft:500}}>
                {AuctionButtion}
                </div>
                <Modal
                    title="确认竞价"
                    visible={this.state.visible}
                    onOk={this.handleOk}
                    onCancel={this.handleCancel}
                >
                    <InputNumber
                        defaultValue={this.state.activity.price}
                        formatter={value => ` ${value}`.replace(/\B(?=(\d{3})+(?!\d))/g, ',')}
                        parser={value => value.replace(/\$\s?|(,*)/g, '')}
                        onChange={this.onChange}
                    />
                </Modal>
            </div>

        );
    }
}

// Integer auctionid,
//     Integer price,
//     Integer userid,
//     String activityIcon
