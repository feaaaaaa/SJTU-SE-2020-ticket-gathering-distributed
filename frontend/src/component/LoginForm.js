import React from 'react';
import { Form, Input, Button } from 'antd';
import {Redirect,Link}  from 'react-router-dom';
import {message} from "antd";
import "../css/Login.css"
import {checkUser} from "../service/userService";
import {userInfo} from "../const/userInfo";
import { UserOutlined, LockOutlined } from '@ant-design/icons';
import {login} from "../service/userService";

const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 16 },
};
const tailLayout = {
    wrapperCol: { offset: 8, span: 16 },
};

export class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            user:null,
            firstLogin:true,
        };
    }
    onFinish = values => {
        // const callback =  (data) => {
        //     this.setState({user:data,firstLogin:false});
        // };
        // checkUser(values,callback);
        // console.log(values);
        if(values.username===userInfo.username && values.password===userInfo.password){
            // this.setState({firstLogin:false,user:userInfo});
            login(values,this);
        }
        this.setState({firstLogin:false});
    };

    onFinishFailed = errorInfo => {
        console.log('Failed:', errorInfo);
    };

    render(){
        if(this.state.user==null &&this.state.firstLogin===false){
            message.error("密码错误")
        }
        if(this.state.user!=null) {
            localStorage.clear();
            // console.log(this.state.user.userId);
            localStorage.setItem("userId", this.state.user.userId);
            localStorage.setItem("username",this.state.user.username);
            return <Redirect to={{pathname: "/"}}/>;
        }
        return (
            <div>
            <Form
                id = "login_form"
                {...layout}
                name="basic"
                initialValues={{remember: true}}
                onFinish={this.onFinish}
                onFinishFailed={this.onFinishFailed}
            >
                <Form.Item
                    // label="Username"
                    // labelCol={{font-size:20px}}
                    name="username"
                    rules={[{required: true, message: 'Please input your username!'}]}
                >
                    <Input prefix={<UserOutlined className="site-form-item-icon" />} placeholder="Username" size="large"
                           style={{fontFamily:"BookMan Old Style",borderRadius: 10,width:400}}/>
                </Form.Item>
                <Form.Item
                    // label="Password"
                    name="password"
                    rules={[{required: true, message: 'Please input your password!'}]}
                >
                    <Input
                        prefix={<LockOutlined className="site-form-item-icon" />}
                        type="password"
                        placeholder="Password"
                        size="large"
                        style={{fontFamily:"BookMan Old Style",borderRadius: 10,width:400}}
                    />
                </Form.Item>
            </Form>

                <div id="buttonDiv">
                    <Button type="primary" htmlType="submit" size="large" id="loginBtn">
                        登 录
                    </Button>
                    <p> </p>
                    <Link to={{pathname: "/register"}}>
                        <Button size="large" id="regBtn">
                            注册
                        </Button>
                    </Link>
                </div>

            </div>
        );
    }
}


