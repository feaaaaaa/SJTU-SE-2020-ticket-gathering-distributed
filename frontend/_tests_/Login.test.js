jest.mock('node-fetch');
import React from 'react'
import {LoginForm} from "../src/component/LoginForm";
import {login} from "../src/service/userService";
import fetch from 'node-fetch';

const {Response} = jest.requireActual('node-fetch');

describe("test loginform",()=>{
    const wrapper=shallow(<LoginForm/>);
    it("test initialized state",()=>{
        expect(wrapper.state().user).toBeNull();
        expect(wrapper.state().firstLogin).toBeTruthy();
    });

    it("test login function",async()=>{
        const json={
            'userId':1,
            'username':'oligei'
        };
        fetch.mockReturnValue(Promise.resolve(new Response(JSON.stringify(json))));
        await login({username:'oligei',password:'123456'},(data)=>{wrapper.setState({user:data,firstLogin:false})});
        expect(wrapper.state().user.userId).toEqual(1);
        expect(wrapper.state().user.username).toEqual('oligei');
    });

});