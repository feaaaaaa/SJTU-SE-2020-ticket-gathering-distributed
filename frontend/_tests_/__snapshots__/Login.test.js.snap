// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test loginform test initialized state 1`] = `
<div>
  <ForwardRef(InternalForm)
    id="login_form"
    initialValues={
      Object {
        "remember": true,
      }
    }
    labelCol={
      Object {
        "span": 8,
      }
    }
    name="basic"
    onFinish={[Function]}
    onFinishFailed={[Function]}
    wrapperCol={
      Object {
        "span": 16,
      }
    }
  >
    <FormItem
      name="username"
      rules={
        Array [
          Object {
            "message": "Please input your username!",
            "required": true,
          },
        ]
      }
    >
      <Input
        placeholder="Username"
        prefix={
          <ForwardRef(UserOutlined)
            className="site-form-item-icon"
          />
        }
        size="large"
        style={
          Object {
            "borderRadius": 10,
            "fontFamily": "BookMan Old Style",
            "width": 400,
          }
        }
        type="text"
      />
    </FormItem>
    <FormItem
      name="password"
      rules={
        Array [
          Object {
            "message": "Please input your password!",
            "required": true,
          },
        ]
      }
    >
      <Input
        placeholder="Password"
        prefix={
          <ForwardRef(LockOutlined)
            className="site-form-item-icon"
          />
        }
        size="large"
        style={
          Object {
            "borderRadius": 10,
            "fontFamily": "BookMan Old Style",
            "width": 400,
          }
        }
        type="password"
      />
    </FormItem>
    <FormItem>
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        id="loginBtn"
        loading={false}
        size="large"
        type="primary"
      >
        登 录
      </Button>
      <p>
         
      </p>
      <Link
        to={
          Object {
            "pathname": "/register",
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          id="regBtn"
          loading={false}
          size="large"
        >
          注册
        </Button>
      </Link>
    </FormItem>
  </ForwardRef(InternalForm)>
  <div
    id="buttonDiv"
  />
</div>
`;

exports[`test loginform test login function 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/",
    }
  }
/>
`;
